src/controllers/NI/src/NIDAQmx.cpp

```diff
@@ -301,6 +301,35 @@ bool NIDAQmx::WriteDigitalLines(
 }
 
 
+/*
+* Get the list of all ni device names as a vector of string
+*/
+std::vector<std::string> NIDAQmx::GetListOfDevices(){
+    constexpr size_t bufferSize = 1000;
+    char devicenamesbuffer[bufferSize] = {};
+
+    if(DAQmxFailed(DAQmxGetSysDevNames(devicenamesbuffer, bufferSize))){
+        spdlog::error("NI device search failed");
+        std::vector<std::string> empty;
+        return empty;
+    }else{
+        std::string tempstring(devicenamesbuffer);
+        std::vector<std::string> devicenamesbuffer;
+        std::stringstream ss("Dev1,Dev2,Dev3");
```

Comment:
are there only ever 3 devices, I don't understand what this is for

Reply:
This is left over from testing. I don't have any devices so I hardcoded what the output would look like to test my code. I updated the code now.